src/Tasks/WriteCodeFragment.cs

```diff
@@ -245,5 +274,312 @@ private string GenerateCode(out string extension)
+                        // we don't use it as a parameter name.
+                        keysToRemove.Add(key);
+
+                        // The parameter will have an explicit type. The metadata value is the type name.
+                        parameterTypes[parameterNameKey] = new ParameterType {
+                            Kind = ParameterTypeKind.Typed,
+                            TypeName = value
+                        };
+                    }
+                }
+                else if (key.EndsWith(IsLiteralSuffix, StringComparison.OrdinalIgnoreCase))
+                {
+                    // Remove the suffix to get the corresponding parameter name.
+                    var parameterNameKey = key.Substring(0, key.Length - IsLiteralSuffix.Length);
+
+                    // As mentioned above for the type name metadata, we will only treat
+                    // this metadata item as a literal flag if there's a corresponding
+                    // metadata item for the parameter for backward-compatibility reasons.
+                    if (customMetadata.Contains(parameterNameKey))
+                    {
+                        // Remove this metadata item so that
+                        // we don't use it as a parameter name.
+                        keysToRemove.Add(key);
+
+                        // If the value is true, the parameter value will be the exact code
+                        // that needs to be written to the generated file for that parameter.
+                        if (string.Equals(value, "true", StringComparison.OrdinalIgnoreCase))
+                        {
+                            parameterTypes[parameterNameKey] = new ParameterType {
+                                Kind = ParameterTypeKind.Literal
+                            };
+                        }
+                    }
+                }
+            }
+
+            // Remove any metadata items that we used
+            // for type names or literal flags.
+            foreach (var key in keysToRemove)
+            {
+                customMetadata.Remove(key);
+            }
+
+            return parameterTypes;
+        }
+
+        /// <summary>
+        /// Uses the given parameters to add CodeDom arguments to the given attribute.
+        /// Returns true if the arguments could be defined, or false if the values could
+        /// not be converted to the required type. An error is also logged for failures.
+        /// </summary>
+        private bool AddArguments(
+            CodeAttributeDeclaration attribute,
+            Lazy<Type> attributeType,
+            IReadOnlyList<AttributeParameter> parameters,
+            bool isPositional
+        )
+        {
+            Type[] constructorParameterTypes = null;
+
+            for (int i = 0; i < parameters.Count; i++)
+            {
+                AttributeParameter parameter = parameters[i];
+                CodeExpression value;
+
+                switch (parameter.Type.Kind)
```

Comment:
I'm not sure this is possible because the case for `ParameterTypeKind.Typed` can return false instead of setting `value`.

Reply:
For the return false case, you had previously called `TryConvertParameterValue(parameter.Type.TypeName, parameter.Value, out value)`, which would set value.

Reply:
The default case causes problems here because it delays finding the positional parameters until they are needed, and that can't (as far as I'm aware) be put in a switch expression. This could be worked around using a `Lazy<Type[]>` for the positional parameter types, but that would result in an allocation that _could_ be avoided in some situations.

I'm happy to go ahead and make that change, but just thought I would check if it's worth the extra allocation. 😄

Reply:
I was thinking something like:
```C#
int t = 0;
char[] b = null;
return t switch
{
        2 => 3,
         _ => otherFunc((b = b is null ? new char[] { 'A', 'B' } : b)[0])
};
```

That's a little messy, but I think it would be cleaner. Probably have to write out everything to see for sure, and if you are happy with it this way, I am, too.